code/web/src/modules/user/UserInfo.js

```diff
@@ -0,0 +1,105 @@
+// Imports
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+
+// UI Imports
+import { H3, H5 } from '../../ui/typography'
+import { Input, File, Textarea } from '../../ui/input'
+
+class UserInfo extends React.Component {
+  constructor() {
+    super() 
+    this.state = {
+      isEditing: false,
+      emailInput: '',
+      userDescription: '',
+      shippingAddressInput: ''
+    }
+  }
+
+  componentDidMount = () => {
+    const user = this.props.user.details
+    this.setState({ 
+      emailInput: user.email,
+      userDescription: user.description || 'Add a description',
+      shippingAddressInput: user.shippingAddress || 'Add your shipping address'
+    })
+  }
+
+  toggleEdit = () => {
+    this.setState({ isEditing: !this.state.isEditing })
+  }
+
+  handleChange = (e) => {
+    const { name, value } = e.target
```

Comment:
Sets local state to reflect changes made in the inputs

Reply:
I like that this one is easy to follow considering the method is called "setState"